regen.py

```diff
@@ -34,40 +33,53 @@ def replace(match: Match[str]) -> str:
     return VARIABLE_RE.sub(replace, tmpl)
 
 
+def is_template_file(filepath: Path) -> bool:
+    return (
+        filepath.suffix == 'in'
+        or filepath.name == '__init__.py'
```

Comment:
If this is to satisfy the example I gave in the issue, oops.  I didn't mean to special-case `__init__.py` -- if we use `__init__.py.in` as the example, then this becomes simpler.

Reply:
Another reason to keep the `.in` requirement would be the `.with_suffix('')` below would also need special-casing otherwise.